### ISSUE TITLE
We need to be able to Add a new Source Reference when adding Physico-chemical data to a site

### ISSUE THREAD

**Issue by helendallas** [kartoza/django-bims#3056]
When physico-chemical data is added to a site, we need to be able to choose or add a new Source Reference.

At the moment, we can only choose a Source Reference. We need to be able to Add a new Source Reference here as well please.

Then Choose or Add Source Reference - we actually also need to be able Add Source Reference here for water temp module as well (I will create another linked ticket)

![image](https://user-images.githubusercontent.com/45286876/166910233-0be54e8c-ed73-491a-b407-7e181102f596.png)

**Comment by helendallas** [comment]
So this isn't working correctly, some issues:

I added Date, Select an existing Source Reference, add data, and submit. It takes me to the validate screen at the end. I think it should rather just show the site? 

https://testing.rbis.kartoza.com/site-visit/list/?validated=[%22non%20validated%22]&o=date

Then when I add the date, and select Add New Source Reference, which I then add, submit, 

The date disappears, as well as any data I have added.
